File: src/gestorAplicacion/Pago.java
package gestorAplicacion;

import java.io.Serializable;

import gestorAplicacion.personas.Medico;

public class Pago implements Serializable{
	private static final long serialVersionUID = 1L;
	private int id;
	private int valor;
	private Consulta consulta;
	private Medico medico;
	private boolean pagado;
	private Entrega entrega;
	
	// Constructor
	public Pago(int valor, boolean pagado) {
		this.setId((int)(Math.random()*10000+1));
		this.setValor(valor);
		this.setPagado(pagado);
		this.setConsulta(null);
	}
	
	public Pago(int valor, Consulta consulta, boolean pagado) {
		this.setId((int)(Math.random()*10000+1));
		this.setValor(valor);
		this.setPagado(pagado);
		this.setConsulta(consulta);
	}
	public Pago(int id, int valor, Medico medico, boolean pagado) {
		this.setId(id);
		this.setValor( valor);
		this.setPagado(pagado);
		this.setMedico(medico);
	}

	// Getters y Setters
	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public int getValor() {
		return valor;
	}

	public void setValor(int valor) {
		this.valor = valor;
	}

	public Consulta getConsulta() {
		return consulta;
	}

	public void setConsulta(Consulta consulta) {
		this.consulta = consulta;
	}

	public boolean isPagado() {
		return pagado;
	}

	public void setPagado(boolean pagado) {
		this.pagado = pagado;
	}

	public Medico getMedico() {
		return medico;
	}

	public void setMedico(Medico medico) {
		this.medico = medico;
	}
}

File: Python/src/gestorAplicacion/Pago.py
"""

"""
import random as rd
class Pago():
    def __init__(self, id = int(rd.random() * 10000 + 1), valor = None, consulta = None, medico = None,
                pagado = False, entrega = None):
        self._id = id
        self._valor = valor
        self._consulta = consulta
        self._medico = medico
        self._pagado = pagado
        self._entrega = entrega
    
    # Getters y Setters
    def getId(self):
        return self._id
    
    def setId(self, id):
        self._id = id
        
    def getValor(self):
        return self._valor
    
    def setValor(self, valor):
        self._valor = valor
        
    def getConsulta(self):
        return self._consulta
    
    def setConsulta(self, consulta):
        self._consulta = consulta
        
    def getMedico(self):
        return self._medico
    
    def setMedico(self, medico):
        self._medico = medico
        
    def isPagado(self):
        return self._pagado
    
    def setPagado(self, pagado):
        self._pagado = pagado
        
    def getEntrega(self):
        return self._entrega
    
    def setEntrega(self, entrega): 
        self._entrega = entrega